api/src/middleware/assessmentsRouter.ts

```diff
@@ -252,7 +255,146 @@ assessmentsRouter.get(
 assessmentsRouter.get(
   '/program/:programAssessmentId/submissions/new',
   async (req, res, next) => {
-    res.json();
+    // get the principal row ID number
+    const { principalId } = req.session;
+
+    // get and parse the program assessment row ID number
+    const { programAssessmentId } = req.params;
+
+    const programAssessmentIdParsed = Number(programAssessmentId);
+
+    if (
+      !Number.isInteger(programAssessmentIdParsed) ||
+      programAssessmentIdParsed < 1
+    ) {
+      next(
+        new BadRequestError(
+          `"${programAssessmentId}" is not a valid program assessment ID.`
+        )
+      );
+      return;
+    }
+
+    try {
+      const programAssessment = await findProgramAssessment(
+        programAssessmentIdParsed
+      );
+
+      if (!programAssessment) {
+        next(
+          new NotFoundError(
+            `Could not find program assessment(ID ${programAssessmentIdParsed}).`
+          )
+        );
+        return;
+      }
+
+      if (new Date(programAssessment.available_after + 'Z') > new Date()) {
```

Comment:
can you explain a little bit what does it mean?

Reply:
JavaScript dates and UTC, I believe!